specs/language/resources.tex

```diff
@@ -0,0 +1,943 @@
+  void doit() {
+    AppendStructuredBuffer lbuf<int3x3> = gbuf;
+  }
+\end{HLSL}
+AppendStructuredBuffer operands to the assignment operator must have the same element type.
+
+\Sub{Dimensions}{Resources.apstbufs.dims}
+
+The structure count and stride of a AppendStructuredBuffer can be retrieved using the \texttt{GetDimensions} method.
+\begin{HLSL}
+void GetDimensions(out uint count, out uint stride);
+\end{HLSL}
+
+This returns number of structured elements of the buffer through the \texttt{count} out parameter
+and the size of each element in bytes through the \texttt{stride} out parameter.
+
+\Sub{Append}{Resources.apstbufs.append}
+
+\begin{HLSL}
+   void Append(in T value )
+\end{HLSL}
+
+Appends the \texttt{value} of the template parameter types to the end of the AppendStructuredBuffer.
+Subsequent appends will continue to add elements to the end of the buffer.
+
+\Sec{ConsumeStructuredBuffer}{Resources.cnstbufs}
+
+ConsumeStructuredBuffers are buffers that can only have values pulled from them as an input stream.
+
+\begin{HLSL}
+template <typename T>
+ class ConsumeStructuredBuffer {
+   ConsumeStructuredBuffer();
+   ConsumeStructuredBuffer operator=(ConsumeStructuredBuffer buf);
+
+   void GetDimensions(out uint count, out uint stride);
+
+   void Consume();
+};
+\end{HLSL}
+
+\Sub{Constructors}{Resources.cnstbufs.ctrs}
+
+ConsumeStructuredBuffers must be defined with explicit template parameter types.
+\begin{HLSL}
+  struct S {int i; float f;};
+  ConsumeStructuredBuffer<S> buf1;
+  ConsumeStructuredBuffer<float4> buf2;
+  ConsumeStructuredBuffer<float2x3> buf3;
+\end{HLSL}
+
+When defined at global scope, ConsumeStructuredBuffers are bound to externally-defined backing stores
+using the explicit binding location if provided or the implicit binding if not.
+
+When defined at local scope, ConsumeStructuredBuffers represent local references
+that can be associated with global buffers when assigned,
+but must be resolvable to a unique global buffer.
+
+\begin{HLSL}
+  ConsumeStructuredBuffer<int3x3> gbuf;
+  void doit() {
+    ConsumeStructuredBuffer lbuf<int3x3> = gbuf;
+  }
+\end{HLSL}
+ConsumeStructuredBuffer operands to the assignment operator must have the same element type.
+
+\Sub{Dimensions}{Resources.cnstbufs.dims}
+
+The structure count and stride of a ConsumeStructuredBuffer can be retrieved using the \texttt{GetDimensions} method.
+\begin{HLSL}
+void GetDimensions(out uint count, out uint stride);
+\end{HLSL}
+
+This returns number of structured elements of the buffer through the \texttt{count} out parameter
+and the size of each element in bytes through the \texttt{stride} out parameter.
+
+\Sub{Consume}{Resources.cnstbufs.consume}
+
+\begin{HLSL}
+   void Consume();
+\end{HLSL}
+
+Removes and returns a value of the template parameter types from the end of the ConsumeStructuredBuffer.
+
+\Sec{Constant Buffers}{Resources.cnbuf}
+
+\Sec{Samplers}{Resources.samp}
+
+
+\Sec{Resource Binding}{Resources.binding}
```

Comment:
This needs to have a discussion about bindings on UDT's.

Reply:
Added